compiler/src/Language/Bond/Codegen/Cpp/ApplyOverloads.hs

```diff
@@ -16,10 +16,8 @@ import Language.Bond.Codegen.Util
 -- | Protocol data type is used to specify what protocols the @Apply@ function
 -- overloads should be generated for.
 data Protocol =
-    Protocol
-    { protocolReader :: String -- ^ Name of the class implementing the protocol reader.
-    , protocolWriter :: String -- ^ Name of the class implementing the protocol writer.
-    }
+    ProtocolReader String | -- ^ Name of the class implementing the protocol reader.
```

Comment:
For the CHANGELOG: this is a major version change for the "bond" Haskell library.